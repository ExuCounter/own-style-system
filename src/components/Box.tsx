import {
  space,
  color,
  borders,
  shadow,
  overflow,
  position,
  background,
  OverflowProps,
  BordersProps,
  ShadowProps,
  SpaceProps,
  ColorProps,
  PositionProps,
  BackgroundProps,
} from 'styled-system'
import { ComponentPropsWithRef, forwardRef } from 'react'
import styled from 'styled-components'

type BoxCoreProps = SpaceProps &
  ColorProps &
  BordersProps &
  ShadowProps &
  OverflowProps &
  PositionProps &
  BackgroundProps

type BoxProps = ComponentPropsWithRef<'div'> & BoxCoreProps

const StyledBox = styled.div<BoxCoreProps>`
  display: block;
  ${space}
  ${color}
  ${borders}
  ${shadow}
  ${overflow}
  ${position}
  ${background}
`

export const Box = forwardRef<HTMLDivElement, BoxProps>((props, ref) => {
  return <StyledBox ref={ref} {...props} />
})

export default Box
