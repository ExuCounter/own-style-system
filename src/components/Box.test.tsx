import TestRenderer from 'react-test-renderer'
import { Box } from './Box'

test('Render correctly Box component', () => {
  const component = TestRenderer.create(<Box>something</Box>).toJSON()
  expect(component).toMatchSnapshot()
})
