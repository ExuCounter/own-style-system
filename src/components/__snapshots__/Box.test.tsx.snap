// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box test 1`] = `
<div
  className="sc-bdfBwQ QQPDI"
>
  something
</div>
`;
